cli/fmt.rs

```diff
@@ -173,6 +189,29 @@ fn format_stdin(check: bool) -> Result<(), ErrBox> {
   Ok(())
 }
 
+async fn run_parallelized<F>(
+  file_paths: Vec<PathBuf>,
+  f: F,
+) -> Result<(), ErrBox>
+where
+  F: FnOnce(PathBuf) -> Result<(), ErrBox> + Send + 'static + Clone,
+{
+  let handles = file_paths.into_iter().map(|file_path| {
+    let f = f.clone();
+    tokio::task::spawn_blocking(move || f(file_path))
+  });
+  let handle_results = futures::future::try_join_all(handles).await.unwrap();
```

Comment:
The `.unwrap()` will panic on the first panic that occurs while formatting. What would probably be more ideal is to do a `join_all` then go over each result to check for panics and then return an error like `"Panic formatting {filePath}: {error}"`. Does doing that sound good? I wanted to get some feedback on the general idea before implementing too much.

Reply:
I prefer panics to terminate all execution with as little code as possible

Reply:
The motivation here would be that if a user has a panic in a certain file it would be nice for them to say "I am getting a panic formatting this file with this code" rather than "I am getting a panic somewhere formatting all my code".

When I was playing around with this I stumbled onto [a panic](https://github.com/dsherret/dprint/issues/183) then I had to go add `println!("{}", file_path_str)` messages in the code to figure out which file had the problem.

Reply:
That would also allow them to quickly add a `// dprint-ignore-file` comment in that file then move on for the time being. In the error message, we might want to suggest they report the bug then add one of those comments for the time being.

Reply:
Just thinking about this more... perhaps I should add catching panics down into dprint itself and returning them as an error on the result? It's rare, but I've found [one scenario](https://github.com/dsherret/dprint/issues/183) where swc panics and it would be nice to display this to the user with dprint's cli as well (so we could both share that code).

I'm not sure what the performance impact is like of having those catch and unwind statements, but I'll find out.

Reply:
I'm fine leaving it crashing - a panic is call that for a reason.

It seems like it would be useful to print the name of the file being formatted just before it jumps into dprint, so if it crashes we'd at least see the filename.

Reply:
Right now this code only prints out formatted files' paths. Printing out every file path would probably slow things down in some terminals (also, with this being parallelized they wouldn't be able to tell exactly which file caused a panic when there's multiple being formatted at the same time).

Reply:
I'm thinking to use [`std::panic::set_hook`](https://doc.rust-lang.org/std/panic/fn.set_hook.html) and [`std::panic::take_hook`](https://doc.rust-lang.org/std/panic/fn.take_hook.html) internally in dprint in order to include the file path in the panic. Anyway, I'll look into all this tomorrow.